src/components/App.svelte

```diff
@@ -109,6 +110,10 @@
 
 			if (text) {
 				$lineData$ = [...$lineData$, { id: generateRandomUUID(), text }];
+
+				if ($maxLines$ > 0 && $lineData$.length > $maxLines$) {
+					$lineData$.splice(0, 1);
```

Comment:
splicing is an inplace operation and would not trigger the correct localstorage update. better to adjust the condition and move it before the new assignment

Other side effects coming to my mind which probably should be adjusted:
- selected lines: would still show the icons as active if the spliced line was selected
- unique lines: would not remove the old value and therefore block any further insertion of the spliced text later on
- undo action: would add lines back, ignoring the setting. for now it would probably be sufficient to just ensure that the list length is not getting over the setting and just allow modification of elements with ids otherwise